src/ami-set.js

```diff
@@ -51,6 +55,49 @@ AmiSet.listAmiSets = async function () {
 };
 
 /**
+ * Checks if the AMIs from the amiSet are valid or not.
+ */
+
+async function checkAmi(ctx, region, ami, vtype) {
+  let request;
+  let missing = [];
+  let virtualizationType = '';
+
+  if (ami) {
+    request = {
+      ImageIds: [ami],
+    };
+    let exists = await amiExists(ctx.ec2[region], ami);
```

Comment:
let's generalise this instead of doing the same describeImages call twice.  No reason to call it once to see if it's there, then again to get more info.  If you make the changes that I suggested to ami-exists.js then this would be a little easier.